include/irr/core/math/aabb2dSIMD.h

```diff
@@ -0,0 +1,169 @@
+#ifndef __IRR_AABB_SIMD_2D_H_INCLUDED__
+#define __IRR_AABB_SIMD_2D_H_INCLUDED__
+
+#include "irr/core/math/irrMath.h"
+#include "irr/core/memory/memory.h"
+#include "irr/core/alloc/AlignedBase.h"
+
+#include <iostream>
+
+#define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
+#define SIGN_FLIP_MASK_XYZW core::vector3du32_SIMD(0x80000000, 0x80000000, 0x80000000, 0x80000000)
+
+
+namespace irr { namespace core {
+
+class aabb2dSIMDf
+{
+public:
+	inline aabb2dSIMDf()
+		: box( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX ) {};
+
+	inline aabb2dSIMDf(const core::vectorSIMDf& _box)
+		: box(_box ^ SIGN_FLIP_MASK_XY) {};
+
+	inline void addBox(const aabb2dSIMDf& _box)		
+	{ 
+		box = core::max_(box, _box.getBounds()); 
+	}
+
+	inline void addPoint(const core::vectorSIMDf& point)	
+	{
+		addBox(point); 
+	}
+
+	inline core::vectorSIMDf getBounds() const
```

Comment:
there's no such function in the original., there shoudl be a separate `getMinEdge` and `getMaxEdge`